From 103ff54f379b7745c9b0a0b29f85cc0f512f70f6 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Tue, 29 Jul 2025 14:45:22 +0200
Subject: [PATCH] Revert "drm/panel: otm8009a: Add default-on property"

This reverts commit cc016885f6dffc8f516676fc5d138d8a337f9e21.
This property was not added in the binding & is no longer needed.

Change-Id: I0b2fe7aad49e4b3241a2dc96a095d41f1f74c4d6
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/479216
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../gpu/drm/panel/panel-orisetech-otm8009a.c  | 30 ++++++++-----------
 1 file changed, 13 insertions(+), 17 deletions(-)

diff --git a/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c b/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
index a360c3b20011..f3b2a57a82af 100644
--- a/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
+++ b/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
@@ -299,6 +299,7 @@ static int otm8009a_unprepare(struct drm_panel *panel)
 	if (!ctx->prepared)
 		return 0;
 
+	pm_runtime_set_autosuspend_delay(ctx->dev, 1000);
 	pm_runtime_mark_last_busy(panel->dev);
 	ret = pm_runtime_put_autosuspend(panel->dev);
 	if (ret < 0)
@@ -446,11 +447,7 @@ static int otm8009a_probe(struct mipi_dsi_device *dsi)
 	if (!ctx)
 		return -ENOMEM;
 
-	if (device_property_read_bool(dev, "default-on"))
-		ctx->reset_gpio = devm_gpiod_get_optional(dev, "reset", GPIOD_OUT_LOW);
-	else
-		ctx->reset_gpio = devm_gpiod_get_optional(dev, "reset", GPIOD_OUT_HIGH);
-
+	ctx->reset_gpio = devm_gpiod_get_optional(dev, "reset", GPIOD_OUT_LOW);
 	if (IS_ERR(ctx->reset_gpio))
 		return dev_err_probe(dev, PTR_ERR(ctx->reset_gpio),
 				     "cannot get reset GPIO\n");
@@ -471,12 +468,17 @@ static int otm8009a_probe(struct mipi_dsi_device *dsi)
 			  MIPI_DSI_MODE_NO_EOT_PACKET;
 
 	pm_runtime_enable(ctx->dev);
-	pm_runtime_set_autosuspend_delay(ctx->dev, 1000);
+	/* set delay to 60s to keep alive the panel to wait the splash screen */
+	pm_runtime_set_autosuspend_delay(ctx->dev, 60000);
 	pm_runtime_use_autosuspend(ctx->dev);
 
 	drm_panel_init(&ctx->panel, dev, &otm8009a_drm_funcs,
 		       DRM_MODE_CONNECTOR_DSI);
 
+	pm_runtime_get_sync(ctx->dev);
+	pm_runtime_mark_last_busy(ctx->dev);
+	pm_runtime_put_autosuspend(ctx->dev);
+
 	ctx->bl_dev = devm_backlight_device_register(dev, dev_name(dev),
 						     dev, ctx,
 						     &otm8009a_backlight_ops,
@@ -484,7 +486,7 @@ static int otm8009a_probe(struct mipi_dsi_device *dsi)
 	if (IS_ERR(ctx->bl_dev)) {
 		ret = PTR_ERR(ctx->bl_dev);
 		dev_err(dev, "failed to register backlight: %d\n", ret);
-		goto disable_pm_runtime;
+		return ret;
 	}
 
 	ctx->bl_dev->props.max_brightness = OTM8009A_BACKLIGHT_MAX;
@@ -498,16 +500,10 @@ static int otm8009a_probe(struct mipi_dsi_device *dsi)
 	if (ret < 0) {
 		dev_err(dev, "mipi_dsi_attach failed. Is host ready?\n");
 		drm_panel_remove(&ctx->panel);
-		goto disable_pm_runtime;
+		return ret;
 	}
 
 	return 0;
-
-disable_pm_runtime:
-	pm_runtime_dont_use_autosuspend(dev);
-	pm_runtime_disable(dev);
-
-	return ret;
 }
 
 static void otm8009a_remove(struct mipi_dsi_device *dsi)
@@ -525,6 +521,9 @@ static __maybe_unused int orisetech_otm8009a_suspend(struct device *dev)
 {
 	struct otm8009a *ctx = dev_get_drvdata(dev);
 
+	gpiod_set_value_cansleep(ctx->reset_gpio, 1);
+	msleep(20);
+
 	regulator_disable(ctx->supply);
 
 	return 0;
@@ -535,9 +534,6 @@ static __maybe_unused int orisetech_otm8009a_resume(struct device *dev)
 	struct otm8009a *ctx = dev_get_drvdata(dev);
 	int ret;
 
-	gpiod_set_value_cansleep(ctx->reset_gpio, 1);
-	msleep(20);
-
 	ret = regulator_enable(ctx->supply);
 	if (ret < 0) {
 		dev_err(ctx->dev, "failed to enable supply: %d\n", ret);
-- 
2.53.0

